util/create_fencing_agent.sh

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env bash
+#
+#######################################################################################################################################################
+#
+# This script simplifies the user interaction with Azure to create a fencing
+# agent service principal and grant the required permissions to manage resources.
+# This script is essentially wrapping the process described here:
+# https://docs.microsoft.com/en-us/azure/virtual-machines/workloads/sap/high-availability-guide-suse-pacemaker#create-azure-fence-agent-stonith-device
+#
+#######################################################################################################################################################
+
+# exit immediately if an unset variable is used
+set -o nounset
+
+# import common functions that are reused across scripts
+source util/common_utils.sh
+
+# name of the script where the auth info will be saved
+readonly auth_script="set-clustering-auth-${1}.sh"
```

Comment:
Maybe consider giving default value here eg:
```suggestion
readonly auth_script="set-clustering-auth-${1-HN1}.sh"
```
Other wise, just run util/create_fencing_agent.sh given an unclear error.
```
:sap-hana nancyc$ util/create_fencing_agent.sh
util/create_fencing_agent.sh: line 19: 1: unbound variable
```

Reply:
The default should be "" (empty string) to avoid the error, but stay consistent with other util script behaviour. Otherwise it should read the SID from the template (although there is no active/specified template in this context).

I gave this feedback previously, but it got missed/regressed.